src/main/java/net/openhft/chronicle/queue/impl/single/TableStoreWriteLock.java

```diff
@@ -88,11 +88,16 @@ public void lock() {
                     "Lock was held by " + lockedBy;
             if (dontRecoverLockTimeout)
                 throw new UnrecoverableTimeoutException(new IllegalStateException(warningMsg + UNLOCK_MAIN_MSG));
-            warn().on(getClass(), warningMsg + ". Unlocking forcibly");
-            forceUnlock(value);
-            // we should reset the pauser after a timeout exception
-            tlPauser.reset();
-            lock();
+            else if (onlyUnlockIfProcessDead) {
+                if (forceUnlockIfProcessIsDead())
+                    lock();
+                else
+                    throw new UnrecoverableTimeoutException(new IllegalStateException(warningMsg + UNLOCK_MAIN_MSG));
+            } else {
+                warn().on(getClass(), warningMsg + UNLOCKING_FORCIBLY_MSG);
+                forceUnlock(value);
```

Comment:
I feel like the value variable could be named better (e.g. currentLockValue or currentPID or whatever it is

Reply:
Now we're increasing the number of lines between where it's assigned and where it's used